R/mime_part.R

```diff
@@ -176,9 +176,26 @@ mime_part.character <- function(x, name, ...) {
   if (length(x) == 1 && file.exists(x)) {
     .file_attachment(x, name, ...)
   } else {
-     .mime_part(headers=list(
-                 "Content-Type"="text/plain",
-                 "Content-Disposition"="inline"),
-               text=paste(x, collapse="\r\n"))
+
+    # Default content_type (backward compatibillity)
+    content_type <- "text/plain"
+
+    # If Encoding is set then use this
+    # R will always use ASCII over UTF-8 if possible
+    # ASCII is reported as unknown
+    encoding <- setdiff(Encoding(x), "unknown")
+
+    if (length(encoding == 1))
+      content_type <- paste0("text/plain; charset=", encoding, "; format=flowed")
+
+    # If all character strings are valid UTF-8 this takes precedent
+    # Normally all character strings in R are either ASCII or UTF-8 which are both valid.
+    if (all(validUTF8(x)))
```

Comment:
Why not convert all strings to UTF-8 if any can't be represented as ASCII? Something like
```R
e <- Encoding(x)
isUTF8 <- validUTF8(x)
if (any(e != "unknown")) {
  x <- iconv(x, ifelse(e == "unknown", ifelse(isUTF8, "utf-8", "ascii"), "utf-8")
  x <- enc2utf8(x) # Really needed?
  content_type <- paste0("text/plain; charset=utf-8")
} else if (all(validUTF8(x))) {
  x <- enc2utf8(x) # Really needed?
  content_type <- paste0("text/plain; charset=utf-8")
} else {
  content_type <- "text/plain"
}
```

Reply:
I also think that converting everything to UTF-8 is best. My old solution did allow for sending e-mails in other encodings, but this could have resulted in problems depending on the OS and locale of the recipient. 

I think using `enc2utf8()` is sufficient, since ASCII is valid UTF-8 and will be reported as unknown by R even after conversion. So, conversion is only needed for other encodings.

I added a separate function to handle conversion and charset setting. This also allows me to better avoid if else chaining :)

Reply:
Perfect, thanks!